src/context.rs

```diff
@@ -111,20 +99,19 @@ fn create_dt(date: NaiveDate, time: navitia_model::objects::Time) -> NaiveDateTi
         + chrono::Duration::seconds(i64::from(time.total_seconds()))
 }
 
-fn create_timetable(ntm: &navitia_model::Model, generation_period: &Period) -> Timetable {
-    info!("computing timetable for {:?}", &generation_period);
+fn create_timetable(ntm: &navitia_model::Model, horizon: &chrono::Duration) -> Timetable {
+    info!("computing timetable for {:?}", &horizon);
     let begin_dt = Utc::now();
     let mut timetable = Timetable {
         connections: vec![],
     };
+
+    let max_date = (chrono::Local::today() + *horizon).naive_local();
+
     for (vj_idx, vj) in ntm.vehicle_journeys.iter() {
         let service = ntm.calendars.get(&vj.service_id).unwrap();
         for st in &vj.stop_times {
-            for date in service
-                .dates
-                .iter()
-                .filter(|date| generation_period.contains(**date))
-            {
+            for date in service.dates.iter().filter(|date| **date <= max_date) {
```

Comment:
we don't want to generate this timetable on past data, do we ?

Reply:
well… it allowed to pass the tests ;) Now the tests are broken :'(